src/main/filesystem.ts

```diff
@@ -83,9 +88,52 @@ export class KangarooFileSystem {
   }
 
   randomPasswordExists() {
-    const pwPath = path.join(this.appDataDir, '.pw');
+    const pwPath = path.join(this.profileDataDir, '.pw');
     return fs.existsSync(pwPath);
   }
+
+  async openLogs() {
+    try {
+      await shell.openPath(this.profileLogsDir);
+    } catch (e) {
+      dialog.showErrorBox('Failed to open logs folder', (e as any).toString());
+    }
+  }
+
+  async exportLogs() {
+    try {
+      const zip = new AdmZip();
+      zip.addLocalFolder(this.profileLogsDir);
+      const exportToPathResponse = await dialog.showSaveDialog({
+        title: 'Export Logs',
+        buttonLabel: 'Export',
+        defaultPath: `Moss_${app.getVersion()}_logs_${Date.now()}.zip`,
+      });
+      if (exportToPathResponse.filePath) {
+        zip.writeZip(exportToPathResponse.filePath);
+        shell.showItemInFolder(exportToPathResponse.filePath);
+      }
+    } catch (e) {
+      dialog.showErrorBox('Failed to export logs', (e as any).toString());
+    }
+  }
+
+  async factoryReset(keepLogs = false) {
+    if (keepLogs) throw new Error('Keeping logs across factory reset is currently not supported.');
+    if (platform.isWindows) {
+      try {
```

Comment:
Should this be a for-try structure so we continue trying to clear more if one step fails?

Reply:
Like this? https://github.com/holochain/kangaroo-electron/pull/45/commits/8dbf38a6f81b80db48bfde30c637c0b71311e7ea